src/autoload.h

```diff
@@ -11,15 +11,19 @@
 #include "common.h"
 #include "lru.h"
 
-// A struct responsible for recording an attempt to access a file.
+/// Recording an attempt to access a file.
 struct file_access_attempt_t {
-    time_t mod_time;      // modification time of the file
-    time_t last_checked;  // when we last checked the file
-    bool accessible;      // whether we believe we could access this file
-    bool stale;           // whether the access attempt is stale
-    int error;            // if we could not access the file, the error code
+    /// modification time of the file
+    time_t mod_time;
```

Comment:
Does your IDE recognize this form:

```
time_t mod_time;  /// modification time of the file
```

If it does that's my preference since it's more compact without sacrificing readability. But if not what you've done here is fine with the caveat that preface comments (as opposed to inline) should generally read as normal sentences; i.e., Begin with a capital letter, end with a period, have proper grammar, etc.